client/state/notices/middleware.js

```diff
@@ -122,6 +122,12 @@ export const onPublicizeConnectionCreate = ( dispatch, { connection } ) => dispa
 	} ) )
 );
 
+export const onPublicizeConnectionCreateFailure = ( dispatch, { error } ) => dispatch(
+	errorNotice( error.message || translate( 'An error occurred while connecting the account.', {
```

Comment:
@gwwar Yes, the error messages come back translated. I'm not sure we can switch based on error codes because there are quite a view codes that have different messages attached to them, including programmatical error messages like 
```php
return new WP_Error( 'connection_error', $publicize_connection_id->get_error_message(), 400 );
```